### PR TITLE
Exception for no detectors

### LINKED ISSUE #332: Assert that there must be at least one detector passed to the fly plan
You can currently pass an empty list into the fly-plan which shouldn't be allowed. We should raise an exception in this case.

### PR DESCRIPTION
**Pull request by abbiemery**
Fixes #332

### CODE REVIEW

File: src/ophyd_async/plan_stubs/fly.py [hunk 1]
```diff
@@ -32,6 +32,10 @@ def time_resolved_fly_and_collect_with_static_seq_table(
     stages/unstages the devices, and opens and closes the run.
 
     """
+    # Ensure there is at least on detector
+    if len(detectors) == 0:
```
Comment (DiamondJoseph) [inline src/ophyd_async/plan_stubs/fly.py:36]: Can also 

`if detectors:` as a list's truthiness is whether it is empty or not, and it catches `detectors = None`

File: src/ophyd_async/plan_stubs/fly.py [hunk 2]
```diff
@@ -32,6 +32,10 @@ def time_resolved_fly_and_collect_with_static_seq_table(
...
+        raise ValueError
```
Comment (DiamondJoseph) [inline src/ophyd_async/plan_stubs/fly.py:37]: Want some detail in here, so you don't have to see the line to understand

File: tests/plan_stubs/test_fly.py [hunk 3]
```diff
@@ -349,3 +349,25 @@ def fly():
         "stream_datum",
         "stop",
     ]
+
+
+async def test_at_least_one_detector_in_fly_plan(
+    RE: RunEngine, flyer, detector_list=[]
```
Comment (DiamondJoseph) [inline tests/plan_stubs/test_fly.py:355]: test detector_list = None as well?

**Review by DiamondJoseph — CHANGES_REQUESTED** [review]

File: src/ophyd_async/plan_stubs/fly.py (hunk 1 above)
Comment (abbiemery) [reply src/ophyd_async/plan_stubs/fly.py:36]: OOOh nice

File: tests/plan_stubs/test_fly.py (hunk 3 above)
Comment (abbiemery) [reply tests/plan_stubs/test_fly.py:355]: Done

Comment (DiamondJoseph) [reply tests/plan_stubs/test_fly.py:355]: Hasn't been pushed if so

Comment (abbiemery) [reply tests/plan_stubs/test_fly.py:355]: Whoops

Comment (abbiemery) [reply tests/plan_stubs/test_fly.py]: Pushed now.

**Review by DiamondJoseph — APPROVED** [review]

**Pull request merged**